test/solidity/Libraries/LibUtil.t.sol

```diff
@@ -90,4 +116,74 @@ contract LibUtilTest is Test {
         vm.expectRevert(revertData);
         mainContract.topLevelFunction2(address(calledContract));
     }
+
+    function test_AddressToBytes32ConversionRoundtrip() public view {
+        // convert initial address to bytes32
+        bytes32 converted = implementer.convertAddressToBytes32(testAddress);
+
+        // convert bytes32 result back to address
+        address addressConverted = implementer.convertBytes32ToAddress(
+            converted
+        );
+
+        // compare results
+        assert(testAddress == addressConverted);
+    }
+
+    function test_Bytes32ToAddressConversionRoundtrip() public view {
+        // [pre-commit-checker: not a secret]
+        bytes32 testAddrBytes32 = hex"0000000000000000000000000000000000000000000000000000000000001234";
+
+        // convert initial address to address
+        address converted = implementer.convertBytes32ToAddress(
+            testAddrBytes32
+        );
+
+        // make sure converted address matches with initial test address
+        assert(converted == testAddress);
+
+        // convert address result back to bytes32
+        bytes32 addressBytes32Converted = implementer.convertAddressToBytes32(
+            converted
+        );
+
+        // compare results
+        assert(testAddrBytes32 == addressBytes32Converted);
+    }
+
+    function test_AddressToBytesMemoryConversionRoundtrip() public view {
+        // convert initial address to bytes memory
+        bytes memory converted = implementer.convertAddressToBytes(
+            testAddress
+        );
+
+        // convert bytes memory result back to address
+        address addressConverted = implementer.convertBytesToAddress(
+            converted
+        );
+
+        // compare results
+        assert(testAddress == addressConverted);
+    }
+
+    function test_BytesMemoryToAddressConversionRoundtrip() public view {
+        bytes memory testAddrBytes = abi.encodePacked(testAddress);
+
+        // convert initial bytes memory address to address
+        address converted = implementer.convertBytesToAddress(testAddrBytes);
+
+        // make sure converted address matches with initial test address
+        assert(converted == testAddress);
+
+        // convert address result back to bytes32
+        bytes memory addressBytesConverted = implementer.convertAddressToBytes(
+            converted
+        );
+
+        // compare results
+        assert(
+            keccak256(abi.encode(testAddrBytes)) ==
+                keccak256(abi.encode(addressBytesConverted))
+        );
+    }
```

Comment:
would be good also to test InvalidCallData revert

Reply:
Could you please elaborate what kind of test you have in mind? 
We do not have any InvalidCallData error in the LibUtil code. 
While we could call the function with some bogus data (that is not picked up by the compiler), what would we expect to get back exactly and how would this help?
@mirooon

Reply:
I think he's referring to line 73 in the lib converting bytes to address where you could pass more than 20 bytes

Reply:
yes!

Reply:
ahhh sorry, got it. I overlooked this one. Will add a test

Reply:
added